### PR TITLE
Standalone endpoints - Phase 1

### PR DESCRIPTION
**Pull request by nb-ohad**
### Explain the changes
- Allow endpoint to run on a separate pod (separate from the core pod/agent pod)
- Remove endpoint related code from Agent
- Remove endpoint related code from Node Monitor
- Fix UI to accommodate above changes

### Issues: Fixed #xxx / Gap #xxx
1. 

### Testing Instructions:
1.

### CODE REVIEW

File: src/server/node_services/nodes_monitor.js [hunk 1]
```diff
@@ -2657,47 +2515,7 @@ class NodesMonitor extends EventEmitter {
...
-        } else if (storage_priority > s3_priority) {
-            host_item.mode = storage_mode === 'OFFLINE' ? 'STORAGE_OFFLINE' : storage_mode;
-        } else {
-            host_item.mode = mode_by_priority[s3_priority];
-            // in case the mode is optimal, check if there are blocked ports
-            if (host_item.mode === 'OPTIMAL' && storage_mode === 'N2N_PORTS_BLOCKED') {
-                host_item.mode = 'N2N_PORTS_BLOCKED';
-            }
-        }
+        host_item.mode = storage_mode;
```
Comment (nimrod-becker) [inline src/server/node_services/nodes_monitor.js]: why did you remove mode_priority ?

Comment (nb-ohad) [reply src/server/node_services/nodes_monitor.js]: After the change, there is no conflict between storage_mode and endpoint_mode because there is no endpoint mode. so the priority of each is not relevant anymore.

File: src/agent/agent.js [hunk 2]
```diff
@@ -1159,27 +1011,13 @@ class Agent {
             });
     }
 
-    set_debug_node(req) {
+    async set_debug_node(req) {
```
[the remark this answers is not in the document]
Comment (nb-ohad) [reply src/agent/agent.js]: Will open after PR is in master

File: src/server/node_services/nodes_monitor.js (hunk 1 above)
Comment (nimrod-becker) [reply src/server/node_services/nodes_monitor.js]: It was also between different drives with different modes.
But for, for now, we don't foresee a multi-drive so ok.

Comment (nb-ohad) [reply src/server/node_services/nodes_monitor.js]: No, it was not used for that. The only use was between the aggr storage nodes mode and the aggr endpoint nodes mode. we still support multi-drive host with a proper mode

File: src/server/system_services/schemas/agent_config_schema.js [hunk 3]
```diff
@@ -31,9 +31,6 @@ module.exports = {
         use_storage: {
             type: 'boolean'
         },
-        use_s3: {
```
Comment (dannyzaken) [inline src/server/system_services/schemas/agent_config_schema.js]: what about upgrades?

Comment (nb-ohad) [reply src/server/system_services/schemas/agent_config_schema.js]: First I don't know if we have deployments with agents or that we need to take them into account.
Second, we can leave the definition. It is optional in any case.
Third, I can  create an upgrade script to delete all agent_configs (or just the field)

@nimrod-becker What od you think

**Review by nimrod-becker — APPROVED** [review]

File: src/server/system_services/schemas/agent_config_schema.js (hunk 3 above)
Comment (nimrod-becker) [reply src/server/system_services/schemas/agent_config_schema.js]: Assume systems with agents

Comment (dannyzaken) [reply src/server/system_services/schemas/agent_config_schema.js]: I don't think we should leave it just to avoid verification errors. just add an upgrade script

File: src/deploy/NVA_build/noobaa_init.sh [hunk 4]
```diff
@@ -151,6 +163,9 @@ then
 elif [ "${RUN_INIT}" == "init_mongo" ]
 then
   init_pod
+elif [ "${RUN_INIT}" == "init_endpoint" ]
```
Comment (guymguym) [inline src/deploy/NVA_build/noobaa_init.sh]: Why not just `endpoint`? Seems like you copied that from `init_mongo` which is just an init container and in the endpoint case this is the actual endpoint run command.

File: src/test/unit_tests/coretest.js [hunk 5]
```diff
@@ -99,13 +99,12 @@ function setup(options = {}) {
     server_rpc.get_base_address = () => 'fcall://fcall';
 
     // register api servers and bg_worker servers locally too
+    // do not regsiter the object, func and common serivces as they are
```
Comment (guymguym) [inline src/test/unit_tests/coretest.js]: I don't understand how this works, isn't these services needed to be available on fcall://fcall for the coretest router?

**Review by guymguym — COMMENTED** [review]
@nb-ohad This PR is really hard to review because it mixes the cleanup of enums and statuses and the new functionality of running the endpoint as standalone.
I would much prefer to review it separately.
I also think the cleanup itself had been better off deferred in a few weeks to make sure the functionality is stable.
That is my 2 cents. Take it anywhere you think from here.

File: src/endpoint/endpoint.js [hunk 6]
```diff
@@ -8,215 +8,108 @@ require('../util/panic');
...
 const ssl_utils = require('../util/ssl_utils');
 const net_utils = require('../util/net_utils');
 const http_utils = require('../util/http_utils');
-
 const s3_rest = require('./s3/s3_rest');
 const blob_rest = require('./blob/blob_rest');
 const lambda_rest = require('./lambda/lambda_rest');
 const { FtpFileSystemNB } = require('./ftp/ftp_filesystem');
+const system_store = require('../server/system_services/system_store').get_instance();
+const md_server = require('../server/md_server');
```
Comment (guymguym) [inline src/endpoint/endpoint.js]: require md_server should change to register_common/object/func_services not in bare module

File: src/endpoint/endpoint.js [hunk 7]
```diff
@@ -8,215 +8,108 @@ require('../util/panic');
...
+const server_rpc = require('../server/server_rpc.js');
+server_rpc.rpc.router = get_router(process.env);
```
Comment (guymguym) [inline src/endpoint/endpoint.js]: move to run_server

File: src/deploy/NVA_build/noobaa_init.sh (hunk 4 above)
Comment (nb-ohad) [reply src/deploy/NVA_build/noobaa_init.sh]: All the functions start with init, init_agent, init_noobaa_server, init_mongo. It describes the fact that the function's code is initializing the environment and then call run_<component>_container

File: src/test/unit_tests/coretest.js (hunk 5 above)
Comment (nb-ohad) [reply src/test/unit_tests/coretest.js]: See my latest commit

File: src/endpoint/endpoint.js (hunk 6 above)
Comment (nb-ohad) [reply src/endpoint/endpoint.js]: Fixed with the last commit

File: src/endpoint/endpoint.js (hunk 7 above)
Comment (nb-ohad) [reply src/endpoint/endpoint.js]: Fixed with the last commit

**Comment by nb-ohad** [comment]
@guymguym the PR is split to commit by area (s3 server, clean up of BE, clean up of FE, fix of tests) to enable reviewers to review by commit instead of review for the whole PR. This makes the review process much easier and more fluent.

**Review by guymguym — APPROVED** [review]
